src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int INITIAL_CAPACITY = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            array = grow();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bounds: " + index);
+        }
+        if (size == array.length) {
+            array = grow();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
```

Comment:
```suggestion
        System.arraycopy(array, index, array, index + 1, size - index);
```
цим можна знехтувати так як ми тут одразу використовуємо arraycopy

Reply:
Тогда не проходит тест checkingResizeInAddByIndex